`server` fails with runtime error when setting logging parameters via environmental variables

relevant version: `v0.1.3`

## What occurs

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x69d098]

goroutine 1 [running]:
github.com/urfave/cli/v2.(*GenericFlag).Apply(0x4000348340, 0xce0730?)
	/go/pkg/mod/github.com/urfave/cli/v2@v2.25.7/flag_generic.go:77 +0xf8
github.com/urfave/cli/v2.flagSet({0xb4d266, 0x8}, {0x40005cd2c0, 0xb, 0x18?}, {{0x0?, 0x400014c2a0?}, 0x0?, 0x0?})
	/go/pkg/mod/github.com/urfave/cli/v2@v2.25.7/flag.go:177 +0x150
github.com/urfave/cli/v2.(*Command).newFlagSet(...)
	/go/pkg/mod/github.com/urfave/cli/v2@v2.25.7/command.go:281
github.com/urfave/cli/v2.(*Command).parseFlags(0x40003cb600, {0xce2638, 0x400014c2a0}, 0xa0?)
	/go/pkg/mod/github.com/urfave/cli/v2@v2.25.7/command.go:314 +0x60
github.com/urfave/cli/v2.(*Command).Run(0x40003cb600, 0x4000548ac0, {0x4000030080, 0x4, 0x4})
	/go/pkg/mod/github.com/urfave/cli/v2@v2.25.7/command.go:155 +0xa8
github.com/urfave/cli/v2.(*App).RunContext(0x4000028000, {0xce03b0?, 0x400003e020}, {0x4000030080, 0x4, 0x4})
	/go/pkg/mod/github.com/urfave/cli/v2@v2.25.7/app.go:332 +0x568
github.com/urfave/cli/v2.(*App).Run(...)
	/go/pkg/mod/github.com/urfave/cli/v2@v2.25.7/app.go:309
github.com/m-mizutani/ghnotify/pkg/controller/cmd.Run({0x4000030080, 0x4, 0x4})
	/src/pkg/controller/cmd/cli.go:204 +0xcb4
main.main()
	/src/main.go:10 +0x34
```

## When it occurs

part of my `docker-compose.yml`:
```
version: '3'
services:
  ghnotify:
    image: ghcr.io/m-mizutani/ghnotify:v0.1.3
    command:
      - --remote-url
      - http://opa_server:8181/v1/data/github/notify
      - serve
    env_file:
      - .env
    environment:
      - GHNOTIFY_ADDR=0.0.0.0:80
      - GHNOTIFY_WEBHOOK_SECRET=my-dummy-secret
      - GHNOTIFY_LOG_LEVEL=debug                    # set log-level from environmental variable
    ports:
      - 4080:80
    networks:
      - my_network
  opa_server
    ...
```

## Note

- The problem seems to be introduced by the following change: https://github.com/m-mizutani/ghnotify/commit/baf08365d80bf4e72b587ec87b11be26128874c6#diff-c476334f550b8032218c65595e89038bb8931089c22ceea3b27d45ce13d40e87L163-R186
- I already fixed the problem on https://github.com/m-mizutani/ghnotify/pull/7

--- a/pkg/controller/cmd/cli.go
+++ b/pkg/controller/cmd/cli.go
@@ -174,7 +174,7 @@ func Run(argv []string) error {
 				Name:        "log-format",
 				Category:    "logging",
 				Aliases:     []string{"f"},
-				Usage:       "Log format [text|json]",
+				Usage:       "Log format [console|json]",
 				EnvVars:     []string{"GHNOTIFY_LOG_FORMAT"},
 				Destination: &logFormat,
 				Value:       &logFormat,

--- a/pkg/controller/cmd/cli.go
+++ b/pkg/controller/cmd/cli.go
@@ -177,6 +177,7 @@ func Run(argv []string) error {
 				Usage:       "Log format [text|json]",
 				EnvVars:     []string{"GHNOTIFY_LOG_FORMAT"},
 				Destination: &logFormat,
+				Value:       &logFormat,
 			},
 			&cli.GenericFlag{
 				Name:        "log-output",
@@ -185,6 +186,7 @@ func Run(argv []string) error {
 				Usage:       "Log output [stdout|stderr]",
 				EnvVars:     []string{"GHNOTIFY_LOG_OUTPUT"},
 				Destination: &logOutput,
+				Value:       &logOutput,
 			},
 		},
 		Commands: []*cli.Command{

--- a/pkg/controller/cmd/cli.go
+++ b/pkg/controller/cmd/cli.go
@@ -168,6 +168,7 @@ func Run(argv []string) error {
 				Usage:       "Log level [debug|info|warn|error]",
 				EnvVars:     []string{"GHNOTIFY_LOG_LEVEL"},
 				Destination: &logLevel,
+				Value:       &logLevel,
 			},
 			&cli.GenericFlag{
 				Name:        "log-format",

--- a/pkg/controller/cmd/cli_test.go
+++ b/pkg/controller/cmd/cli_test.go
@@ -1,6 +1,7 @@
 package cmd_test
 
 import (
+	"fmt"
 	"os"
 	"testing"
 	"time"
@@ -10,60 +11,59 @@ import (
 )
 
 func TestRun(t *testing.T) {
-	tests := []struct {
-		name     string
-		envKey   string
-		envValue string
-		port     string
-	}{
-		{
-			name:     "configure using GHNOTIFY_LOG_LEVEL",
-			envKey:   "GHNOTIFY_LOG_LEVEL",
-			envValue: "debug",
-			port:     "4080",
+	type EnvTest struct {
+		port  string
+		value string
+	}
+
+	testCases := map[string]map[string]EnvTest{
+		"GHNOTIFY_LOG_LEVEL": {
+			"debug": EnvTest{"4081", "debug"},
+			"info":  EnvTest{"4082", "info"},
+			"warn":  EnvTest{"4083", "warn"},
+			"error": EnvTest{"4084", "error"},
 		},
-		{
-			name:     "configure using GHNOTIFY_LOG_FORMAT",
-			envKey:   "GHNOTIFY_LOG_FORMAT",
-			envValue: "json",
-			port:     "4081",
+		"GHNOTIFY_LOG_FORMAT": {
+			"text": EnvTest{"4085", "text"},
+			"json": EnvTest{"4086", "json"},
 		},
-		{
-			name:     "configure using GHNOTIFY_LOG_OUTPUT",
-			envKey:   "GHNOTIFY_LOG_OUTPUT",
-			envValue: "stdout",
-			port:     "4082",
+		"GHNOTIFY_LOG_OUTPUT": {
+			"stdout": EnvTest{"4087", "stdout"},
+			"stderr": EnvTest{"4088", "stderr"},
 		},
 	}
 
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			os.Setenv(tt.envKey, tt.envValue)
-			defer os.Unsetenv(tt.envKey)
+	for envKey, variations := range testCases {
+		for variationName, variation := range variations {
+			testName := fmt.Sprintf("%s as %s", envKey, variationName)
+			t.Run(testName, func(t *testing.T) {
+				os.Setenv(envKey, variation.value)
+				defer os.Unsetenv(envKey)
 
-			errCh := make(chan error)
+				errCh := make(chan error)
 
-			go func() {
-				argv := []string{
-					"ghnotify",
-					"--slack-api-token",
-					"dummy",
-					"--remote-url",
-					"localhost:1234",
-					"serve",
-					"--addr",
-					"0.0.0.0:" + tt.port,
+				go func() {
+					argv := []string{
+						"ghnotify",
+						"--slack-api-token",
+						"dummy",
+						"--remote-url",
+						"localhost:1234",
+						"serve",
+						"--addr",
+						"0.0.0.0:" + variation.port,
+					}
+					err := cmd.Run(argv)
+					errCh <- err
+				}()
+
+				select {
+				case err := <-errCh:
+					assert.NoError(t, err)
+				case <-time.After(time.Second * 1):
+					t.Log("cmd.Run exited without error")
 				}
-				err := cmd.Run(argv)
-				errCh <- err
-			}()
-
-			select {
-			case err := <-errCh:
-				assert.NoError(t, err)
-			case <-time.After(time.Second * 1):
-				t.Log("cmd.Run exited without error")
-			}
-		})
+			})
+		}
 	}
 }

--- a/pkg/controller/cmd/cli_test.go
+++ b/pkg/controller/cmd/cli_test.go
@@ -10,30 +10,60 @@ import (
 )
 
 func TestRun(t *testing.T) {
-	t.Run("configure using environmental variable", func(t *testing.T) {
-		os.Setenv("GHNOTIFY_LOG_LEVEL", "debug")
-		defer os.Unsetenv("GHNOTIFY_LOG_LEVEL")
+	tests := []struct {
+		name     string
+		envKey   string
+		envValue string
+		port     string
+	}{
+		{
+			name:     "configure using GHNOTIFY_LOG_LEVEL",
+			envKey:   "GHNOTIFY_LOG_LEVEL",
+			envValue: "debug",
+			port:     "4080",
+		},
+		{
+			name:     "configure using GHNOTIFY_LOG_FORMAT",
+			envKey:   "GHNOTIFY_LOG_FORMAT",
+			envValue: "json",
+			port:     "4081",
+		},
+		{
+			name:     "configure using GHNOTIFY_LOG_OUTPUT",
+			envKey:   "GHNOTIFY_LOG_OUTPUT",
+			envValue: "stdout",
+			port:     "4082",
+		},
+	}
 
-		errCh := make(chan error)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			os.Setenv(tt.envKey, tt.envValue)
+			defer os.Unsetenv(tt.envKey)
 
-		go func() {
-			argv := []string{
-				"ghnotify",
-				"--slack-api-token",
-				"dummy",
-				"--remote-url",
-				"localhost:1234",
-				"serve",
+			errCh := make(chan error)
+
+			go func() {
+				argv := []string{
+					"ghnotify",
+					"--slack-api-token",
+					"dummy",
+					"--remote-url",
+					"localhost:1234",
+					"serve",
+					"--addr",
+					"0.0.0.0:" + tt.port,
+				}
+				err := cmd.Run(argv)
+				errCh <- err
+			}()
+
+			select {
+			case err := <-errCh:
+				assert.NoError(t, err)
+			case <-time.After(time.Second * 1):
+				t.Log("cmd.Run exited without error")
 			}
-			err := cmd.Run(argv)
-			errCh <- err
-		}()
-
-		select {
-		case err := <-errCh:
-			assert.NoError(t, err)
-		case <-time.After(time.Duration(0.1 * float64(time.Second))):
-			t.Log("cmd.Run exited without error")
-		}
-	})
+		})
+	}
 }

--- a/pkg/controller/cmd/cli_test.go
+++ b/pkg/controller/cmd/cli_test.go
@@ -1,0 +1,39 @@
+package cmd_test
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/m-mizutani/ghnotify/pkg/controller/cmd"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRun(t *testing.T) {
+	t.Run("configure using environmental variable", func(t *testing.T) {
+		os.Setenv("GHNOTIFY_LOG_LEVEL", "debug")
+		defer os.Unsetenv("GHNOTIFY_LOG_LEVEL")
+
+		errCh := make(chan error)
+
+		go func() {
+			argv := []string{
+				"ghnotify",
+				"--slack-api-token",
+				"dummy",
+				"--remote-url",
+				"localhost:1234",
+				"serve",
+			}
+			err := cmd.Run(argv)
+			errCh <- err
+		}()
+
+		select {
+		case err := <-errCh:
+			assert.NoError(t, err)
+		case <-time.After(time.Duration(0.1 * float64(time.Second))):
+			t.Log("cmd.Run exited without error")
+		}
+	})
+}

--- a/pkg/controller/cmd/cli_test.go
+++ b/pkg/controller/cmd/cli_test.go
@@ -60,7 +60,7 @@ func TestRun(t *testing.T) {
 				select {
 				case err := <-errCh:
 					assert.NoError(t, err)
-				case <-time.After(time.Second * 1):
+				case <-time.After(time.Duration(0.01 * float64(time.Second))):
 					t.Log("cmd.Run exited without error")
 				}
 			})

--- a/pkg/controller/cmd/cli_test.go
+++ b/pkg/controller/cmd/cli_test.go
@@ -24,7 +24,7 @@ func TestRun(t *testing.T) {
 			"error": EnvTest{"4084", "error"},
 		},
 		"GHNOTIFY_LOG_FORMAT": {
-			"text": EnvTest{"4085", "text"},
+			"text": EnvTest{"4085", "console"},
 			"json": EnvTest{"4086", "json"},
 		},
 		"GHNOTIFY_LOG_OUTPUT": {

